portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -28,18 +31,30 @@
   private final String NAME = "name"; 
   @Override
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    String comment = getParameter(request, COMMENT, "");
+    String user = getParameter(request, NAME, "");
+    long timestamp = System.currentTimeMillis();
+    
+    Entity commentEntity = new Entity("comment");
+    commentEntity.setProperty(NAME, user);
+    commentEntity.setProperty(COMMENT, comment);
+    commentEntity.setProperty("timestamp", timestamp);
+
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    datastore.put(commentEntity);
+    
+    //send appropriate confirmation response
     String finalResponse = "";
-    String getMessage = getParameter(request, COMMENT, "");
     
     response.setContentType("text/html;");
 
-    if (getMessage.equals("")) {
+    if (comment.equals("")) {
       finalResponse = "Got no comment.";
       response.getWriter().println("<h3>" + finalResponse + "</h3>");
-    }else {
-      finalResponse = "Submitted! Thank you " + getParameter(request, NAME, "");
+    } else {
+      finalResponse = "Submitted! Thank you " + user;
```

Comment:
Nit: when user is equal to empty string, the returned message will end with an extra space. This is no big deal for now, but good to keep corner cases in mind.